mpdev/internal/apply/deployment_manager_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apply
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/exec"
+	testingexec "k8s.io/utils/exec/testing"
+)
+
+func TestDeploymentManager(t *testing.T) {
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+
+	testcases := []struct {
+		name            string
+		expectedRunArgs [][]string
+		zipFilePath     string
+		missingRef      bool
+		badRefType      bool
+	}{{
+		name: "Deployment Manager GCS",
+		expectedRunArgs: [][]string{
+			{"zip", "-r", "/tmp/outdir/dm_template.zip", "."},
+			{"gsutil", "cp", "/tmp/outdir/dm_template.zip", "gs://project/dmtemppath.zip"},
+		},
+		zipFilePath: "gs://project/dmtemppath.zip",
+	}, {
+		name: "Deployment Manager Local Save Relative Path",
+		expectedRunArgs: [][]string{
+			{"zip", "-r", filepath.Join(wd, "resourcedir/dir2/localzippath.zip"), "."},
+		},
+		zipFilePath: "dir2/localzippath.zip",
+	},
+		{
+			name: "Deployment Manager Local Save Absolute Path",
+			expectedRunArgs: [][]string{
+				{"zip", "-r", "/tmp/dir3/localzippath.zip", "."},
+			},
+			zipFilePath: "/tmp/dir3/localzippath.zip",
+		},
+		{
+			name:            "Deployment Manager Missing Reference",
+			expectedRunArgs: nil,
```

Comment:
this this explicitness required? Should be `nil` if not specified, right?

Reply:
Fixed